src/mame/drivers/gamate.cpp

```diff
@@ -150,6 +152,7 @@ void gamate_state::init_gamate()
 
 void gamate_state::machine_start()
 {
+	memset(m_ram, 0xff, 0x400);  /* memory seems to contain 0xff at power up */
```

Comment:
Can you get the size of the memory share?  I think `required_shared_ptr` has a `bytes()` member for doing that.  magic numbers tend to be fragile.

Reply:
I've changed it. Thanks for the hint.